corehq/motech/models.py

```diff
@@ -153,8 +153,29 @@ def get_requests(
         )
 
     def get_auth_manager(self):
+        # Auth types that don't require a username:
         if self.auth_type is None:
             return AuthManager()
+        if self.auth_type == OAUTH1:
+            return OAuth1Manager(
+                client_id=self.client_id,
+                client_secret=self.plaintext_client_secret,
+                api_endpoints=self._get_oauth1_api_endpoints(),
+                connection_settings=self,
+            )
+        if self.auth_type == OAUTH2_CLIENT:
+            return OAuth2ClientGrantManager(
+                self.url,
+                client_id=self.client_id,
+                client_secret=self.plaintext_client_secret,
+                token_url=self.token_url,
+                refresh_url=self.refresh_url,
+                connection_settings=self,
+            )
+
+        # Auth types that require a username:
+        if not isinstance(self.username, str):
+            raise ValueError(f'Auth type {self.auth_type!r} requires a username')
```

Comment:
Is it possible that `ConnectionSettings` exist in production databases that will now raise an error here? It seems like it would be a problem if that happened. What do you think of changing this until the `CHECK` constraint solution can be implemented (with accompanying migration to fix bad rows, assuming that is possible)?
```suggestion
            return AuthManager()
```

Reply:
Although those `ConnectionsSettings` would never have worked, you're right, it is possible that they exist (and almost guaranteed to exist in a QA domain on Staging). I'll change this.